src/main/java/org/broadinstitute/hellbender/tools/exome/segmentation/ClusteringGenomicHMMSegmenter.java

```diff
@@ -0,0 +1,305 @@
+package org.broadinstitute.hellbender.tools.exome.segmentation;
+
+import com.google.common.annotations.VisibleForTesting;
+import org.apache.commons.math3.analysis.integration.SimpsonIntegrator;
+import org.apache.commons.math3.analysis.integration.UnivariateIntegrator;
+import org.apache.commons.math3.special.Gamma;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.broadinstitute.hellbender.tools.exome.ModeledSegment;
+import org.broadinstitute.hellbender.utils.*;
+import org.broadinstitute.hellbender.utils.hmm.ForwardBackwardAlgorithm;
+import org.broadinstitute.hellbender.utils.hmm.ViterbiAlgorithm;
+
+import java.util.*;
+import java.util.function.Function;
+import java.util.stream.IntStream;
+
+/**
+ * Segmentation and parameter learning corresponding to ClusteringGenomicHMM
+ *
+ * @author David Benjamin &lt;davidben@broadinstitute.org&gt;
+ */
+public abstract class ClusteringGenomicHMMSegmenter<T> {
+    protected final Logger logger = LogManager.getLogger(ClusteringGenomicHMM.class);
+
+    private double concentration;
+    protected double[] weights; //one per hidden state
+    protected double[] hiddenStateValues;
+    protected double memoryLength;
+    private boolean parametersHaveBeenLearned = false;
+
+    protected static final int NEUTRAL_VALUE_INDEX = 0;
+
+    private static final int MAX_EM_ITERATIONS = 50;
+    protected final List<T> data;
+    protected final List<SimpleInterval> positions;
+    private final double[] distances;   //distances[n] is the n to n+1 distance
+
+    protected static double NEGLIGIBLE_POSTERIOR_FOR_M_STEP = 0.001;
+
+    //private static final Logger logger = LogManager.getLogger(AlleleFractionSegmenter.class);
+    private static final double MINIMUM_MEMORY_LENGTH = 1;
+    private static final double MAXIMUM_MEMORY_LENGTH = 1e10;
+    private static final double DEFAULT_MEMORY_LENGTH = 5e7;
+
+    //paraeters for pruning unused hidden states
+    final double DISTANCE_TO_NEIGHBOR_TO_BE_CONSIDERED_SPURIOUS = 0.01;  //if a hidden state is this close to another state, it might be false
```

Comment:
should this be a constant?  isn't it dependent on the scale of the hidden states?